### PR TITLE
Microsoft.Compute: Add new stable API version 2021-03-01 for CloudServices resource

### PR DESCRIPTION
**Pull request by stephenhou**
### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (RamyasreeChakka) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:64]: Please specify x-ms-long-running-operation-options as well.

Comment (RamyasreeChakka) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:64]: Refer http://azure.github.io/autorest/extensions/#x-ms-long-running-operation-options

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (RamyasreeChakka) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:305]: Add x-ms-long-running-operation-options

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (RamyasreeChakka) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2617]: Similar to other location resource types in compute, I guess locations/cloudServiceOsFamilies resource also uses ProxyOnly, LocationBased routing. Property 'location' cannot be used as a top level property for proxy resources.
Refer the NOTE section in the RP documentation here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (RamyasreeChakka) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2521]: Similar to other location resource types in compute, I guess locations/cloudServiceOsVersions resource also uses ProxyOnly, LocationBased routing. Property 'location' cannot be used as a top level property for proxy resources.
Refer the NOTE section in the RP documentation here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

Comment (stephenhou) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2521]: I see that the note says that location is not applicable in the request body. However, these ProxyOnly resources are only available through GET. Additionally readonly: true is specified. So location would only be applied for the response body in this situation.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (stephenhou) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2617]: I see that the note says that location is not applicable in the request body. However, these ProxyOnly resources are only available through GET. Additionally readonly: true is specified. So location would only be applied for the response body in this situation.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (stephenhou) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:64]: this doesn't seem like a strict requirement (should have, not must have). We have been adding several x-ms-long-running-operation in several APIs without options without this being a concern (is this new?). In our case, the default value used (azure-async-operation) is exactly what we want.

If possible, I would strongly prefer to take an item to make this change in the future as it will require some thought in our swagger generation code.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (stephenhou) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:305]: this doesn't seem like a strict requirement (should have, not must have). We have been adding several x-ms-long-running-operation in several APIs without options without this being a concern (is this new?). In our case, the default value used (azure-async-operation) is exactly what we want.

If possible, I would strongly prefer to take an item to make this change in the future as it will require some thought in our swagger generation code.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (RamyasreeChakka) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2521]: As per offline discussion, as cloudServiceOsVersions are associated with a location, including location in response sounds reasonable.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (RamyasreeChakka) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2617]: As per offline discussion, as cloudServiceOsFamilies can differ based on a location, including location in response sounds reasonable.

**Comment by raych1** [comment]
@stephenhou , regard to `invalid_type` error in model validation, please suppress it in readme.md coz it doesn't allow to represent the file format in example file.

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
Please fix the model validation so that we can unblock the merging

**Comment by stephenhou** [comment]
> @stephenhou , regard to `invalid_type` error in model validation, please suppress it in readme.md coz it doesn't allow to represent the file format in example file.

I wasn't able to suppress the invalid_type error for some reason. But I removed the example and suppressed the error for missing example instead

**Comment by stephenhou** [comment]
> Please fix the model validation so that we can unblock the merging

@ArcturusZhang Done. please merge

**Comment by Sandido** [comment]
I will be reviewing this today.

**Review by ArcturusZhang — APPROVED** [review]

File: specification/compute/resource-manager/readme.md
Comment (Sandido) [inline specification/compute/resource-manager/readme.md:37]: If you want to include this in the 2021-03-01 release, then you should make this PR against the branch https://github.com/Azure/azure-rest-api-specs/tree/compute-2021-03

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2094]: Adding a required property to an existing object is a breaking change. We do not want to have any breaking changes comparing the latest Preview release with the first Stable release. 
I think you're also trying to edit this in the preview version. That Preview needs to release and release the SDK before this change can be merged.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2131]: Adding a required property to an existing object is a breaking change. We do not want to have any breaking changes comparing the latest Preview release with the first Stable release.
I think you're also trying to edit this in the preview version. That Preview needs to release and release the SDK before this change can be merged.

**Review by Sandido — CHANGES_REQUESTED** [review]

**Review by Sandido — APPROVED** [review]

File: specification/compute/resource-manager/readme.md
Comment (stephenhou) [reply specification/compute/resource-manager/readme.md:37]: resolved as per email discussion

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (stephenhou) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2131]: resolved as per email discussion

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (stephenhou) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2094]: resolved as per email discussion

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json
Comment (shnaya434) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json:2149]: Update the description to add the bold text:   "description": "The id reference of the cloud service containing the target IP with which the subject cloud service can perform a swap. This property cannot be updated **or deleted** once it is set. The swappable cloud service referred by this id must be present otherwise an error will be thrown."
        }

**Pull request merged**
